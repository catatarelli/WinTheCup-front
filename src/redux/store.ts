import {
  configureStore,
  type ThunkAction,
  type Action,
} from "@reduxjs/toolkit";
import { uiReducer } from "./features/uiSlice/uiSlice";
import { userReducer } from "./features/user/userSlice";

export const store = configureStore({
  reducer: {
    ui: uiReducer,
    user: userReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
