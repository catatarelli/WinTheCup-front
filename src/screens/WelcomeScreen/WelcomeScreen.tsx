/* eslint-disable react/no-unescaped-entities */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import {
  ImageBackground,
  SafeAreaView,
  ScrollView,
  Text,
  TouchableOpacity,
  View,
} from "react-native";
import React from "react";
import image from "../../../assets/background-image.jpg";
import styles from "../RegisterScreen/RegisterScreenStyled";
import formStyles from "../../styles/form.styles";
import welcomeStyles from "./WelcomeScreenStyled";

const WelcomeScreen = () => (
  <ScrollView>
    <ImageBackground
      source={image}
      resizeMode="cover"
      style={styles.image}
      testID="backgroudImage"
    >
      <SafeAreaView style={welcomeStyles.container}>
        <Text style={welcomeStyles.title}>WIN THE CUP</Text>
        <Text style={welcomeStyles.secondaryTitle}>WORLD CUP PREDICTOR</Text>
        <Text style={welcomeStyles.complementryTitle}>
          WHO'S YOUR CHAMPION?
        </Text>
        <TouchableOpacity style={welcomeStyles.button}>
          <Text
            style={formStyles.buttonText}
            onPress={() => {}}
            testID="loginButton"
          >
            Log in
          </Text>
        </TouchableOpacity>
        <View style={formStyles.loginContainer}>
          <Text style={formStyles.loginText}>Don't have an account?</Text>
          <TouchableOpacity onPress={() => {}}>
            <Text style={formStyles.loginLink}>Join now</Text>
          </TouchableOpacity>
        </View>
      </SafeAreaView>
    </ImageBackground>
  </ScrollView>
);

export default WelcomeScreen;
