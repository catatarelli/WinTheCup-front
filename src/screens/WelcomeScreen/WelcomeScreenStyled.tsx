import { StyleSheet } from "react-native";
import colors from "../../styles/colors.styles";
import formStyles from "../../styles/form.styles";

const welcomeStyles = StyleSheet.create({
  container: {
    display: "flex",
    alignItems: "center",
  },
  logo: {
    marginTop: 140,
    marginBottom: 100,
    width: 350,
    height: 50,
  },
  secondaryTitle: {
    color: colors.white,
    fontSize: 30,
    marginVertical: 20,
    fontWeight: "bold",
  },
  complementryTitle: {
    color: colors.white,
    fontSize: 24,
    marginVertical: 10,
    fontWeight: "bold",
  },
  button: {
    ...formStyles.button,
    marginTop: 80,
    opacity: 0.7,
  },
});

export default welcomeStyles;
