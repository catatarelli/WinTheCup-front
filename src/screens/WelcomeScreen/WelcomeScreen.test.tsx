/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import { render, screen } from "@testing-library/react-native";
import WelcomeScreen from "./WelcomeScreen";
import { Provider } from "react-redux";
import { store } from "../../redux/store";

const mockRegisterUser = jest.fn();

jest.mock("../../hooks/useUser/useUser", () => () => ({
  registerUser: mockRegisterUser,
}));

describe("Given a RegisterScreen component", () => {
  describe("When it's rendered", () => {
    test("Then it should show a background image and the title 'WIN THE CUP'", async () => {
      const imageId = "backgroudImage";
      const expectedText = "WIN THE CUP";

      render(
        <Provider store={store}>
          <WelcomeScreen />
        </Provider>
      );

      const displayedImage = screen.queryByTestId(imageId);
      const title = await screen.getByText(expectedText);

      expect(displayedImage).toBeDefined();
      expect(title).toBeDefined();
    });
  });
});
