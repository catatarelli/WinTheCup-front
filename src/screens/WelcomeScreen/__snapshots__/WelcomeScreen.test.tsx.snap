// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a WelcomeScreen component Then it should match the snapshot 1`] = `
<RCTScrollView>
  <View>
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": 1334,
          "justifyContent": "center",
          "resizeMode": "cover",
          "width": 750,
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 1334,
              "width": 750,
            },
            undefined,
          ]
        }
        testID="backgroudImage"
      />
      <RCTSafeAreaView
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 46,
              "marginBottom": 100,
              "marginTop": 140,
              "width": 350,
            }
          }
          testID="logo"
        />
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 30,
              "fontWeight": "bold",
              "marginVertical": 20,
            }
          }
        >
          WORLD CUP PREDICTOR
        </Text>
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 24,
              "fontWeight": "bold",
              "marginVertical": 10,
            }
          }
        >
          WHO'S YOUR CHAMPION?
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 20,
              "color": "#fff",
              "height": 57,
              "justifyContent": "center",
              "marginTop": 80,
              "marginVertical": 25,
              "opacity": 0.7,
              "padding": 6,
              "width": 266,
              "zIndex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 30,
                "fontWeight": "500",
                "zIndex": 1,
              }
            }
            testID="loginButton"
          >
            Log in
          </Text>
        </View>
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 20,
                "marginRight": 15,
              }
            }
          >
            Don't have an account?
          </Text>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Join now
            </Text>
          </View>
        </View>
      </RCTSafeAreaView>
    </View>
  </View>
</RCTScrollView>
`;
